.github/workflows/test-build.yml

```diff
@@ -0,0 +1,56 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  # Cancel any running workflow for the same branch when new commits are pushed.
+  # We group both by ref_name (available when CI is triggered by a push to a branch/tag)
+  # and head_ref (available when CI is triggered by a PR).
+  group: "${{ github.ref_name }}-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21.x'
+      - name: Go mod check
+        run: make mod-check
+      - name: Lint
+        run: make lint
+      - name: Test benchmarks
+        run: make test-benchmarks
+      - name: Check protos
+        run: |
+          apt-get update && apt-get -y install unzip
+          go mod vendor
+          make check-protos
+
+  test:
+    name: Test on Go ${{ matrix.go-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
It might be worth adding a comment here to explain why we run the tests on both Go versions.

Reply:
Updated in 485a1db93